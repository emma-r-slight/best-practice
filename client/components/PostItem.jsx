import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import { receivePost } from "../actions/posts";

const PostItem = (props) => {
  const { post, receivePost } = props;
  return (
    <li data-testid="post">
      <Link to={`/post/${post.id}`} onClick={() => receivePost(post)}>
        {post.name}
      </Link>
    </li>
  );
};

const mapDispatchToProps = { receivePost };

export default connect(null, mapDispatchToProps)(PostItem);
