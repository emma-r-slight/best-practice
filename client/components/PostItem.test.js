import React from 'react'
import { render, screen } from '@testing-library/react'
import { renderWithRedux } from '../testing/utils'
import '@testing-library/jest-dom/extend-expect'
import '@babel/polyfill'

import PostItem from './PostItem'
import mockPosts from '../testing/mockPosts'

test('<PostItem> includes name in <li>', async () => {
  renderWithRedux(<PostItem post={mockPosts[0]}/>)
  const post = await screen.findByText('mocked post 1')
  expect(post).toBeInTheDocument()
  expect(post).toMatchSnapshot()
})
