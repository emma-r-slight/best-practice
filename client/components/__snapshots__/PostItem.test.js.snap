// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostItem> includes name in <li> 1`] = `
<a
  href="/post/1"
>
  mocked post 1
</a>
`;
